--- origsrc/qt-x11-opensource-src-4.4.0/src/corelib/global/qglobal.h	2008-04-28 08:11:18.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/corelib/global/qglobal.h	2008-07-13 14:15:37.156250000 -0500
@@ -825,7 +825,7 @@
    Workaround for static const members on MSVC++.
 */
 
-#if defined(Q_CC_MSVC)
+#if defined(Q_CC_MSVC) || defined(Q_OS_CYGWIN)
 #  define QT_STATIC_CONST static
 #  define QT_STATIC_CONST_IMPL
 #else
--- origsrc/qt-x11-opensource-src-4.4.0/src/corelib/kernel/qmetaobject.cpp	2008-04-28 08:11:18.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/corelib/kernel/qmetaobject.cpp	2008-07-13 17:28:13.171875000 -0500
@@ -498,7 +498,7 @@
         if (strcmp(self->d.stringdata, name) == 0)
             return self;
         if (self->d.extradata) {
-            const QMetaObject **e = self->d.extradata;
+            QT_STATIC_CONST_IMPL QMetaObject **e = self->d.extradata;
             while (*e) {
                 if (const QMetaObject *m =QMetaObject_findMetaObject((*e), name))
                     return m;
--- origsrc/qt-x11-opensource-src-4.4.0/src/corelib/kernel/qobject.h	2008-04-28 08:11:18.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/corelib/kernel/qobject.h	2008-07-13 17:05:02.250000000 -0500
@@ -281,7 +281,7 @@
 protected:
     QObjectData *d_ptr;
 
-    static const QMetaObject staticQtMetaObject;
+    QT_STATIC_CONST QMetaObject staticQtMetaObject;
 
     friend struct QMetaObject;
     friend class QApplication;
--- origsrc/qt-x11-opensource-src-4.4.0/src/corelib/kernel/qobjectdefs.h	2008-04-28 08:11:18.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/corelib/kernel/qobjectdefs.h	2008-07-13 17:20:54.671875000 -0500
@@ -149,7 +149,7 @@
 #define Q_OBJECT \
 public: \
     Q_OBJECT_CHECK \
-    static const QMetaObject staticMetaObject; \
+    QT_STATIC_CONST QMetaObject staticMetaObject; \
     virtual const QMetaObject *metaObject() const; \
     virtual void *qt_metacast(const char *); \
     QT_TR_FUNCTIONS \
@@ -160,7 +160,7 @@
 /* tmake ignore Q_GADGET */
 #define Q_GADGET \
 public: \
-    static const QMetaObject staticMetaObject; \
+    QT_STATIC_CONST QMetaObject staticMetaObject; \
 private:
 #else // Q_MOC_RUN
 #define slots slots
@@ -408,7 +408,7 @@
         const QMetaObject *superdata;
         const char *stringdata;
         const uint *data;
-        const QMetaObject **extradata;
+        QT_STATIC_CONST_IMPL QMetaObject **extradata;
     } d;
 };
 
--- origsrc/qt-x11-opensource-src-4.4.0/src/dbus/qdbusabstractadaptor.cpp	2008-04-28 08:11:18.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/dbus/qdbusabstractadaptor.cpp	2008-07-13 18:16:18.765625000 -0500
@@ -312,7 +312,7 @@
 // the relaySlot slot has local ID 0 (we use this when calling QMetaObject::connect)
 // it also gets called with the void** array
 
-static const uint qt_meta_data_QDBusAdaptorConnector[] = {
+QT_STATIC_CONST uint qt_meta_data_QDBusAdaptorConnector[] = {
  // content:
        1,       // revision
        0,       // classname
@@ -331,13 +331,13 @@
        0        // eod
 };
 
-static const char qt_meta_stringdata_QDBusAdaptorConnector[] = {
+QT_STATIC_CONST char qt_meta_stringdata_QDBusAdaptorConnector[] = {
     "QDBusAdaptorConnector\0\0obj,metaobject,sid,args\0"
     "relaySignal(QObject*,const QMetaObject*,int,QVariantList)\0\0relaySlot()\0"
     "polish()\0"
 };
 
-const QMetaObject QDBusAdaptorConnector::staticMetaObject = {
+QT_STATIC_CONST_IMPL QMetaObject QDBusAdaptorConnector::staticMetaObject = {
     { &QObject::staticMetaObject, qt_meta_stringdata_QDBusAdaptorConnector,
       qt_meta_data_QDBusAdaptorConnector, 0 }
 };
--- origsrc/qt-x11-opensource-src-4.4.0/src/tools/moc/generator.cpp	2008-04-28 08:11:20.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/tools/moc/generator.cpp	2008-07-13 16:45:06.312500000 -0500
@@ -194,7 +194,7 @@
     qualifiedClassNameIdentifier.replace(':', '_');
 
     int index = 10;
-    fprintf(out, "static const uint qt_meta_data_%s[] = {\n", qualifiedClassNameIdentifier.constData());
+    fprintf(out, "QT_STATIC_CONST uint qt_meta_data_%s[] = {\n", qualifiedClassNameIdentifier.constData());
     fprintf(out, "\n // content:\n");
     fprintf(out, "    %4d,       // revision\n", 1);
     fprintf(out, "    %4d,       // classname\n", strreg(cdef->qualified));
@@ -248,7 +248,7 @@
 //
 // Build stringdata array
 //
-    fprintf(out, "static const char qt_meta_stringdata_%s[] = {\n", qualifiedClassNameIdentifier.constData());
+    fprintf(out, "QT_STATIC_CONST char qt_meta_stringdata_%s[] = {\n", qualifiedClassNameIdentifier.constData());
     fprintf(out, "    \"");
     int col = 0;
     int len = 0;
@@ -302,7 +302,7 @@
         }
     }
     if (!extraList.isEmpty()) {
-        fprintf(out, "static const QMetaObject *qt_meta_extradata_%s[] = {\n    ", qualifiedClassNameIdentifier.constData());
+        fprintf(out, "QT_STATIC_CONST QMetaObject *qt_meta_extradata_%s[] = {\n    ", qualifiedClassNameIdentifier.constData());
         for (int i = 0; i < extraList.count(); ++i) {
             if (i)
                 fprintf(out, ",\n    ");
@@ -317,9 +317,9 @@
 //
 
     if (isQt)
-        fprintf(out, "const QMetaObject QObject::staticQtMetaObject = {\n");
+        fprintf(out, "QT_STATIC_CONST_IMPL QMetaObject QObject::staticQtMetaObject = {\n");
     else
-        fprintf(out, "const QMetaObject %s::staticMetaObject = {\n", cdef->qualified.constData());
+        fprintf(out, "QT_STATIC_CONST_IMPL QMetaObject %s::staticMetaObject = {\n", cdef->qualified.constData());
 
     if (isQObject)
         fprintf(out, "    { 0, ");
--- origsrc/qt-x11-opensource-src-4.4.0/tools/porting/src/errors.cpp	2008-04-28 08:11:21.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/tools/porting/src/errors.cpp	2008-07-13 22:30:55.625000000 -0500
@@ -46,8 +46,8 @@
 
 QT_BEGIN_NAMESPACE
 
-QT_STATIC_CONST_IMPL Error& Errors::InternalError = Error( 1, -1, QLatin1String("Internal Error") );
-QT_STATIC_CONST_IMPL Error& Errors::SyntaxError = Error( 2, -1, QLatin1String("Syntax Error before '%1'") );
-QT_STATIC_CONST_IMPL Error& Errors::ParseError = Error( 3, -1, QLatin1String("Parse Error before '%1'") );
+const Error& Errors::InternalError = Error( 1, -1, QLatin1String("Internal Error") );
+const Error& Errors::SyntaxError = Error( 2, -1, QLatin1String("Syntax Error before '%1'") );
+const Error& Errors::ParseError = Error( 3, -1, QLatin1String("Parse Error before '%1'") );
 
 QT_END_NAMESPACE
--- origsrc/qt-x11-opensource-src-4.4.0/tools/porting/src/errors.h	2008-04-28 08:11:21.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/tools/porting/src/errors.h	2008-07-13 22:30:42.656250000 -0500
@@ -63,9 +63,9 @@
 class Errors
 {
 public:
-    QT_STATIC_CONST Error& InternalError;
-    QT_STATIC_CONST Error& SyntaxError;
-    QT_STATIC_CONST Error& ParseError;
+    static const Error& InternalError;
+    static const Error& SyntaxError;
+    static const Error& ParseError;
 };
 
 QT_END_NAMESPACE
--- origsrc/qt-x11-opensource-src-4.4.0/tools/qdbus/qdbuscpp2xml/qdbuscpp2xml.cpp	2008-04-28 08:11:21.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/tools/qdbus/qdbuscpp2xml/qdbuscpp2xml.cpp	2008-07-21 11:44:02.937500000 -0500
@@ -251,7 +251,7 @@
 
     while (!input->atEnd()) {
         QByteArray line = readLine();
-        if (line.startsWith("static const uint qt_meta_data_")) {
+        if (line.startsWith("QT_STATIC_CONST uint qt_meta_data_")) {
             // start of new class data
             uint *data;
             loadIntData(data);
@@ -261,7 +261,7 @@
                 line = readLine();
                 if (input->atEnd())
                     parseError();
-            } while (!line.startsWith("static const char qt_meta_stringdata_"));
+            } while (!line.startsWith("QT_STATIC_CONST char qt_meta_stringdata_"));
 
             char *stringdata;
             loadStringData(stringdata);
