--- origsrc/qt-x11-opensource-src-4.4.0/tools/assistant/lib/fulltextsearch/qclucene-config_p.h	2008-04-28 08:11:20.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/tools/assistant/lib/fulltextsearch/qclucene-config_p.h	2008-07-13 22:10:18.421875000 -0500
@@ -406,12 +406,14 @@
 #define _CL_HAVE_WCSSTR  1 
 #endif
 
+#if !defined(__CYGWIN__)
 /* Define to 1 if you have the `wcstod' function. */
 #ifndef _CL_HAVE_WCSTOD 
 #define _CL_HAVE_WCSTOD  1 
 #endif
+#endif
 
-#if !defined(__SUNPRO_CC) && !defined(__SUNPRO_C) && !defined(__HP_aCC)
+#if !defined(__SUNPRO_CC) && !defined(__SUNPRO_C) && !defined(__HP_aCC) && !defined(__CYGWIN__)
     /* Define to 1 if you have the `wcstoll' function. */
 #   ifndef _CL_HAVE_WCSTOLL 
 #   define _CL_HAVE_WCSTOLL  1 
