--- origsrc/qt-x11-opensource-src-4.4.0/src/3rdparty/webkit/JavaScriptCore/kjs/DateMath.h	2008-04-28 08:11:16.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/3rdparty/webkit/JavaScriptCore/kjs/DateMath.h	2008-07-13 14:20:03.859375000 -0500
@@ -101,7 +101,7 @@
         , year(inTm.tm_year)
         , isDST(inTm.tm_isdst)
     {
-#if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS_OS) && !PLATFORM(HPUX_OS)
+#if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS_OS) && !PLATFORM(HPUX_OS) && !PLATFORM(CYGWIN)
         utcOffset = static_cast<int>(inTm.tm_gmtoff);
 
         int inZoneSize = strlen(inTm.tm_zone) + 1;
@@ -128,7 +128,7 @@
         ret.tm_year  =  year;
         ret.tm_isdst =  isDST;
 
-#if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS_OS) && !PLATFORM(HPUX_OS)
+#if !PLATFORM(WIN_OS) && !PLATFORM(SOLARIS_OS) && !PLATFORM(HPUX_OS) && !PLATFORM(CYGWIN)
         ret.tm_gmtoff = static_cast<long>(utcOffset);
         ret.tm_zone = timeZone;
 #endif
--- origsrc/qt-x11-opensource-src-4.4.0/src/3rdparty/webkit/JavaScriptCore/kjs/collector.cpp	2008-04-28 08:11:16.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/3rdparty/webkit/JavaScriptCore/kjs/collector.cpp	2008-07-13 14:21:31.250000000 -0500
@@ -46,7 +46,7 @@
 
 #include "CollectorHeapIntrospector.h"
 
-#elif PLATFORM(WIN_OS)
+#elif PLATFORM(WIN_OS) || PLATFORM(CYGWIN)
 
 #include <windows.h>
 
@@ -126,7 +126,7 @@
 #if PLATFORM(DARWIN)    
     vm_address_t address = 0;
     vm_map(current_task(), &address, BLOCK_SIZE, BLOCK_OFFSET_MASK, VM_FLAGS_ANYWHERE, MEMORY_OBJECT_NULL, 0, FALSE, VM_PROT_DEFAULT, VM_PROT_DEFAULT, VM_INHERIT_DEFAULT);
-#elif PLATFORM(WIN_OS)
+#elif PLATFORM(WIN_OS) || PLATFORM(CYGWIN)
      // windows virtual address granularity is naturally 64k
     LPVOID address = VirtualAlloc(NULL, BLOCK_SIZE, MEM_COMMIT | MEM_RESERVE, PAGE_READWRITE);
 #elif HAVE(POSIX_MEMALIGN)
@@ -164,7 +164,7 @@
 {
 #if PLATFORM(DARWIN)    
     vm_deallocate(current_task(), reinterpret_cast<vm_address_t>(block), BLOCK_SIZE);
-#elif PLATFORM(WIN_OS)
+#elif PLATFORM(WIN_OS) || PLATFORM(CYGWIN)
     VirtualFree(block, BLOCK_SIZE, MEM_RELEASE);
 #elif HAVE(POSIX_MEMALIGN)
     free(block);
@@ -291,7 +291,7 @@
 #elif PLATFORM(WIN_OS) && PLATFORM(X86_64) && COMPILER(MSVC)
     PNT_TIB64 pTib = reinterpret_cast<PNT_TIB64>(NtCurrentTeb());
     return (void*)pTib->StackBase;
-#elif PLATFORM(WIN_OS) && PLATFORM(X86) && COMPILER(GCC)
+#elif (PLATFORM(WIN_OS) || PLATFORM(CYGWIN)) && PLATFORM(X86) && COMPILER(GCC)
     // offset 0x18 from the FS segment register gives a pointer to
     // the thread information block for the current thread
     NT_TIB* pTib;
--- origsrc/qt-x11-opensource-src-4.4.0/src/3rdparty/webkit/JavaScriptCore/wtf/Platform.h	2008-04-28 08:11:16.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/3rdparty/webkit/JavaScriptCore/wtf/Platform.h	2008-07-13 14:23:29.171875000 -0500
@@ -50,6 +50,13 @@
 #define WTF_PLATFORM_WIN_OS 1
 #endif
 
+/* PLATFORM(CYGWIN) */
+/* Operating system level dependencies for Cygwin that should be used */
+/* regardless of operating environment */
+#if defined(__CYGWIN__)
+#define WTF_PLATFORM_CYGWIN 1
+#endif
+
 /* PLATFORM(SOLARIS_OS) */
 /* Operating system level dependencies for Solaris that should be used */
 /* regardless of operating environment */
@@ -70,6 +77,7 @@
    || defined(__unix)      \
    || defined(__unix__)    \
    || defined (__NetBSD__) \
+   || defined(__CYGWIN__)  \
    || defined(_AIX)
 #define WTF_PLATFORM_UNIX 1
 #endif
