--- origsrc/qt-x11-opensource-src-4.4.0/mkspecs/cygwin-g++/qmake.conf	2008-04-28 08:11:14.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/mkspecs/cygwin-g++/qmake.conf	2008-07-13 17:30:33.828125000 -0500
@@ -21,7 +21,7 @@
 QMAKE_CFLAGS_WARN_OFF	= -w
 QMAKE_CFLAGS_RELEASE	= -O2
 QMAKE_CFLAGS_DEBUG	= -g
-QMAKE_CFLAGS_SHLIB	=
+QMAKE_CFLAGS_SHLIB	= -DPIC -DDLL_EXPORT
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_THREAD	= -D_REENTRANT
 
@@ -38,23 +38,23 @@
 
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib
+QMAKE_INCDIR_X11	=
+QMAKE_LIBDIR_X11	=
 QMAKE_INCDIR_QT		= $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT		= $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL	=
+QMAKE_LIBDIR_OPENGL	=
 
 QMAKE_LINK		= g++
 QMAKE_LINK_SHLIB	= g++
-QMAKE_LFLAGS		=
+QMAKE_LFLAGS		= -Wl,--enable-auto-import
 QMAKE_LFLAGS_RELEASE	=
 QMAKE_LFLAGS_DEBUG	=
-QMAKE_LFLAGS_SHLIB	= -shared
+QMAKE_LFLAGS_SHLIB	= -shared -Wl,--enable-auto-image-base
 QMAKE_LFLAGS_PLUGIN	= $$QMAKE_LFLAGS_SHLIB
-QMAKE_LFLAGS_SONAME	= -Wl,-soname,
+QMAKE_LFLAGS_SONAME	=
 QMAKE_LFLAGS_THREAD	=
-QMAKE_RPATH		= -Wl,-rpath,
+QMAKE_RPATH		=
 QMAKE_CYGWIN_SHLIB	= 1
 QMAKE_CYGWIN_EXE	= 1
  
@@ -71,7 +71,7 @@
 
 QMAKE_AR		= ar cqs
 QMAKE_OBJCOPY           = objcopy
-QMAKE_RANLIB		=
+QMAKE_RANLIB		= ranlib
 
 QMAKE_TAR		= tar -cf
 QMAKE_GZIP		= gzip -9f
--- origsrc/qt-x11-opensource-src-4.4.0/mkspecs/cygwin-g++/qplatformdefs.h	2008-04-28 08:11:14.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/mkspecs/cygwin-g++/qplatformdefs.h	2008-07-13 15:20:13.875000000 -0500
@@ -30,9 +30,8 @@
 
 #include <sys/types.h>
 #include <sys/ioctl.h>
-// Cygwin does not provide <sys/ipc.h> and <sys/shm.h> because it
-// doesn't support SysV IPC or shared memory. See for example:
-// 	http://afni.nimh.nih.gov/afni/afniboard/messages/1725.html
+#include <sys/ipc.h>
+#include <sys/shm.h>
 #include <sys/time.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
--- origsrc/qt-x11-opensource-src-4.4.0/mkspecs/features/uitools.prf	2008-04-28 08:11:14.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/mkspecs/features/uitools.prf	2008-07-15 02:17:10.187500000 -0500
@@ -7,6 +7,6 @@
     mac: QTUITOOLS_LINKAGE = -lQtUiTools_debug
     win32: QTUITOOLS_LINKAGE = -lQtUiToolsd
 }
-LIBS += $$QTUITOOLS_LINKAGE
+LIBS = $$QTUITOOLS_LINKAGE $$LIBS
 
 INCLUDEPATH = $$QMAKE_INCDIR_QT/QtUiTools $$INCLUDEPATH
