--- origsrc/qt-x11-opensource-src-4.4.0/src/corelib/plugin/qlibrary.cpp	2008-04-28 08:11:18.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/corelib/plugin/qlibrary.cpp	2008-07-13 14:14:07.015625000 -0500
@@ -510,7 +510,7 @@
  */
 bool QLibrary::isLibrary(const QString &fileName)
 {
-#if defined(Q_OS_WIN32) || defined(Q_OS_WINCE)
+#if defined(Q_OS_WIN32) || defined(Q_OS_WINCE) || defined(Q_OS_CYGWIN)
     return fileName.endsWith(QLatin1String(".dll"));
 #else
     QString completeSuffix = QFileInfo(fileName).completeSuffix();
--- origsrc/qt-x11-opensource-src-4.4.0/src/corelib/plugin/qlibrary_unix.cpp	2008-04-28 08:11:18.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/corelib/plugin/qlibrary_unix.cpp	2008-07-13 14:12:27.531250000 -0500
@@ -152,6 +152,13 @@
 # endif
 #elif defined(Q_OS_AIX)
         suffixes << ".a";
+#elif defined(Q_OS_CYGWIN)
+        prefixes << QLatin1String("cyg");
+        if (!fullVersion.isEmpty()) {
+            suffixes << QString::fromLatin1("-%1.dll").arg(fullVersion);
+        } else {
+            suffixes << QLatin1String(".dll");
+        }
 #else
         if (!fullVersion.isEmpty()) {
             suffixes << QString::fromLatin1(".so.%1").arg(fullVersion);
