--- origsrc/qt-x11-opensource-src-4.4.0/config.tests/unix/odbc/odbc.cpp	2008-04-28 08:11:12.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/config.tests/unix/odbc/odbc.cpp	2008-07-13 14:02:58.281250000 -0500
@@ -1,3 +1,7 @@
+#ifdef __CYGWIN__
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#endif
 #include <sql.h>
 #include <sqlext.h>
 
--- origsrc/qt-x11-opensource-src-4.4.0/config.tests/unix/odbc/odbc.pro	2008-04-28 08:11:12.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/config.tests/unix/odbc/odbc.pro	2008-07-13 14:03:25.218750000 -0500
@@ -1,4 +1,5 @@
 SOURCES = odbc.cpp
 CONFIG -= qt dylib
 mac:CONFIG -= app_bundle
-LIBS += -lodbc
+cygwin-g++:LIBS += -lodbc32
+!cygwin-g++:LIBS += -lodbc
--- origsrc/qt-x11-opensource-src-4.4.0/src/plugins/sqldrivers/odbc/odbc.pro	2008-04-28 08:11:19.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/plugins/sqldrivers/odbc/odbc.pro	2008-07-13 14:15:37.171875000 -0500
@@ -11,7 +11,10 @@
 }
 
 unix {
-	!contains( LIBS, .*odbc.* ) {
+	cygwin-g++:!contains( LIBS, .*odbc.* ) {
+   		LIBS    *= -lodbc32
+	}
+	!cygwin-g++:!contains( LIBS, .*odbc.* ) {
 	    LIBS 	*= -lodbc
 	}
 }
--- origsrc/qt-x11-opensource-src-4.4.0/src/sql/drivers/odbc/qsql_odbc.h	2008-04-28 08:11:19.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/sql/drivers/odbc/qsql_odbc.h	2008-07-13 14:15:37.218750000 -0500
@@ -51,6 +51,11 @@
 #include <QtCore/qt_windows.h>
 #endif
 
+#if defined(Q_OS_CYGWIN)
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#endif
+
 #if defined (Q_OS_MAC) && (MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_3)
 // assume we use iodbc on MACX
 // comment next line out if you use a
