--- origsrc/qt-x11-opensource-src-4.5.1/mkspecs/cygwin-g++/qmake.conf	2009-04-21 18:57:37.000000000 -0500
+++ src/qt-x11-opensource-src-4.5.1/mkspecs/cygwin-g++/qmake.conf	2009-04-29 22:22:59.076500000 -0500
@@ -38,25 +38,25 @@ QMAKE_CXXFLAGS_THREAD	= $$QMAKE_CFLAGS_T
 
 QMAKE_INCDIR		=
 QMAKE_LIBDIR		=
-QMAKE_INCDIR_X11	= /usr/X11R6/include
-QMAKE_LIBDIR_X11	= /usr/X11R6/lib
+QMAKE_INCDIR_X11	=
+QMAKE_LIBDIR_X11	=
 QMAKE_INCDIR_QT		= $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT		= $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL	=
+QMAKE_LIBDIR_OPENGL	=
 
-QMAKE_LINK		= g++
-QMAKE_LINK_SHLIB	= g++
-QMAKE_LINK_C		= gcc
-QMAKE_LINK_C_SHLIB	= gcc
+QMAKE_LINK		= g++-4
+QMAKE_LINK_SHLIB	= g++-4
+QMAKE_LINK_C		= gcc-4
+QMAKE_LINK_C_SHLIB	= gcc-4
 QMAKE_LFLAGS		=
 QMAKE_LFLAGS_RELEASE	=
 QMAKE_LFLAGS_DEBUG	=
 QMAKE_LFLAGS_SHLIB	= -shared
 QMAKE_LFLAGS_PLUGIN	= $$QMAKE_LFLAGS_SHLIB
-QMAKE_LFLAGS_SONAME	= -Wl,-soname,
+QMAKE_LFLAGS_SONAME	=
 QMAKE_LFLAGS_THREAD	=
-QMAKE_RPATH		= -Wl,-rpath,
+QMAKE_RPATH		=
 QMAKE_CYGWIN_SHLIB	= 1
 QMAKE_CYGWIN_EXE	= 1
  
@@ -73,7 +73,7 @@ QMAKE_UIC		= $$[QT_INSTALL_BINS]/uic
 
 QMAKE_AR		= ar cqs
 QMAKE_OBJCOPY           = objcopy
-QMAKE_RANLIB		=
+QMAKE_RANLIB		= ranlib
 
 QMAKE_TAR		= tar -cf
 QMAKE_GZIP		= gzip -9f
--- origsrc/qt-x11-opensource-src-4.5.1/mkspecs/cygwin-g++/qplatformdefs.h	2009-04-21 18:57:37.000000000 -0500
+++ src/qt-x11-opensource-src-4.5.1/mkspecs/cygwin-g++/qplatformdefs.h	2009-04-29 22:19:00.107750000 -0500
@@ -71,9 +71,8 @@
 
 #include <sys/types.h>
 #include <sys/ioctl.h>
-// Cygwin does not provide <sys/ipc.h> and <sys/shm.h> because it
-// doesn't support SysV IPC or shared memory. See for example:
-// 	http://afni.nimh.nih.gov/afni/afniboard/messages/1725.html
+#include <sys/ipc.h>
+#include <sys/shm.h>
 #include <sys/time.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
--- origsrc/qt-x11-opensource-src-4.5.1/mkspecs/features/uitools.prf	2009-04-21 18:57:37.000000000 -0500
+++ src/qt-x11-opensource-src-4.5.1/mkspecs/features/uitools.prf	2009-04-29 22:19:56.623375000 -0500
@@ -7,6 +7,6 @@ CONFIG(debug, debug|release) {
     mac: QTUITOOLS_LINKAGE = -lQtUiTools_debug
     win32: QTUITOOLS_LINKAGE = -lQtUiToolsd
 }
-LIBS += $$QTUITOOLS_LINKAGE
+LIBS = $$QTUITOOLS_LINKAGE $$LIBS
 
 INCLUDEPATH = $$QMAKE_INCDIR_QT/QtUiTools $$INCLUDEPATH
