--- origsrc/qt-x11-opensource-src-4.5.2/src/corelib/codecs/qtextcodec.cpp	2009-06-28 01:02:24.552706500 -0500
+++ src/qt-x11-opensource-src-4.5.2/src/corelib/codecs/qtextcodec.cpp	2009-06-28 14:15:46.041072100 -0500
@@ -525,7 +525,7 @@ static void setupLocaleMapper()
     localeMapper = QTextCodec::codecForName("System");
 #else
 
-#ifndef QT_NO_ICONV
+#if !defined(QT_NO_ICONV) && !defined(Q_OS_CYGWIN)
     localeMapper = QTextCodec::codecForName("System");
 #endif
 
