--- origsrc/qt-x11-opensource-src-4.5.1/src/gui/styles/gtksymbols_p.h	2009-04-21 18:57:41.000000000 -0500
+++ src/qt-x11-opensource-src-4.5.1/src/gui/styles/gtksymbols_p.h	2009-05-27 13:45:57.479891000 -0500
@@ -58,6 +58,10 @@
 
 #undef signals // Collides with GTK stymbols
 #include <gtk/gtk.h>
+#ifdef G_PLATFORM_WIN32
+#undef gtk_init
+#endif
+
 #include <QtCore/QLibrary>
 #include <QtGui/QFont>
 #include <QtGui/QFileDialog>
--- origsrc/qt-x11-opensource-src-4.5.1/src/gui/styles/gtksymbols.cpp	2009-04-21 18:57:41.000000000 -0500
+++ src/qt-x11-opensource-src-4.5.1/src/gui/styles/gtksymbols.cpp	2009-06-02 03:27:58.739508100 -0500
@@ -208,6 +208,8 @@ static QString classPath(GtkWidget *widg
 static void resolveGtk()
 {
     const QString GTK_PATH(QLS("gtk-x11-2.0"));
+    const QString GDK_PATH(QLS("gdk-x11-2.0"));
+    const QString GDK_PIXBUF_PATH(QLS("gdk_pixbuf-2.0"));
     QGtk::gtk_init = (Ptr_gtk_init)QLibrary::resolve(GTK_PATH, 0, "gtk_init");
     QGtk::gtk_window_new = (Ptr_gtk_window_new)QLibrary::resolve(GTK_PATH, 0, "gtk_window_new");
     QGtk::gtk_style_attach = (Ptr_gtk_style_attach)QLibrary::resolve(GTK_PATH, 0, "gtk_style_attach");
@@ -228,20 +230,20 @@ static void resolveGtk()
     QGtk::gtk_dialog_run = (Ptr_gtk_dialog_run)QLibrary::resolve(GTK_PATH, 0, "gtk_dialog_run");
     QGtk::gtk_file_chooser_set_filename = (Ptr_gtk_file_chooser_set_filename)QLibrary::resolve(GTK_PATH, 0, "gtk_file_chooser_set_filename");
 
-    QGtk::gdk_pixbuf_get_pixels = (Ptr_gdk_pixbuf_get_pixels)QLibrary::resolve(GTK_PATH, 0, "gdk_pixbuf_get_pixels");
-    QGtk::gdk_pixbuf_get_width = (Ptr_gdk_pixbuf_get_width)QLibrary::resolve(GTK_PATH, 0, "gdk_pixbuf_get_width");
-    QGtk::gdk_pixbuf_get_height = (Ptr_gdk_pixbuf_get_height)QLibrary::resolve(GTK_PATH, 0, "gdk_pixbuf_get_height");
-    QGtk::gdk_pixmap_new = (Ptr_gdk_pixmap_new)QLibrary::resolve(GTK_PATH, 0, "gdk_pixmap_new");
-    QGtk::gdk_pixbuf_new = (Ptr_gdk_pixbuf_new)QLibrary::resolve(GTK_PATH, 0, "gdk_pixbuf_new");
-    QGtk::gdk_pixbuf_get_from_drawable = (Ptr_gdk_pixbuf_get_from_drawable)QLibrary::resolve(GTK_PATH, 0, "gdk_pixbuf_get_from_drawable");
-    QGtk::gdk_draw_rectangle = (Ptr_gdk_draw_rectangle)QLibrary::resolve(GTK_PATH, 0, "gdk_draw_rectangle");
-    QGtk::gdk_pixbuf_unref = (Ptr_gdk_pixbuf_unref)QLibrary::resolve(GTK_PATH, 0, "gdk_pixbuf_unref");
-    QGtk::gdk_drawable_unref = (Ptr_gdk_drawable_unref)QLibrary::resolve(GTK_PATH, 0, "gdk_drawable_unref");
-    QGtk::gdk_drawable_get_depth = (Ptr_gdk_drawable_get_depth)QLibrary::resolve(GTK_PATH, 0, "gdk_drawable_get_depth");
-    QGtk::gdk_color_free = (Ptr_gdk_color_free)QLibrary::resolve(GTK_PATH, 0, "gdk_color_free");
-    QGtk::gdk_x11_window_set_user_time = (Ptr_gdk_x11_window_set_user_time)QLibrary::resolve(GTK_PATH, 0, "gdk_x11_window_set_user_time");
-    QGtk::gdk_x11_drawable_get_xid = (Ptr_gdk_x11_drawable_get_xid)QLibrary::resolve(GTK_PATH, 0, "gdk_x11_drawable_get_xid");
-    QGtk::gdk_x11_drawable_get_xdisplay = (Ptr_gdk_x11_drawable_get_xdisplay)QLibrary::resolve(GTK_PATH, 0, "gdk_x11_drawable_get_xdisplay");
+    QGtk::gdk_pixbuf_get_pixels = (Ptr_gdk_pixbuf_get_pixels)QLibrary::resolve(GDK_PIXBUF_PATH, 0, "gdk_pixbuf_get_pixels");
+    QGtk::gdk_pixbuf_get_width = (Ptr_gdk_pixbuf_get_width)QLibrary::resolve(GDK_PIXBUF_PATH, 0, "gdk_pixbuf_get_width");
+    QGtk::gdk_pixbuf_get_height = (Ptr_gdk_pixbuf_get_height)QLibrary::resolve(GDK_PIXBUF_PATH, 0, "gdk_pixbuf_get_height");
+    QGtk::gdk_pixmap_new = (Ptr_gdk_pixmap_new)QLibrary::resolve(GDK_PATH, 0, "gdk_pixmap_new");
+    QGtk::gdk_pixbuf_new = (Ptr_gdk_pixbuf_new)QLibrary::resolve(GDK_PIXBUF_PATH, 0, "gdk_pixbuf_new");
+    QGtk::gdk_pixbuf_get_from_drawable = (Ptr_gdk_pixbuf_get_from_drawable)QLibrary::resolve(GDK_PIXBUF_PATH, 0, "gdk_pixbuf_get_from_drawable");
+    QGtk::gdk_draw_rectangle = (Ptr_gdk_draw_rectangle)QLibrary::resolve(GDK_PATH, 0, "gdk_draw_rectangle");
+    QGtk::gdk_pixbuf_unref = (Ptr_gdk_pixbuf_unref)QLibrary::resolve(GDK_PIXBUF_PATH, 0, "gdk_pixbuf_unref");
+    QGtk::gdk_drawable_unref = (Ptr_gdk_drawable_unref)QLibrary::resolve(GDK_PATH, 0, "gdk_drawable_unref");
+    QGtk::gdk_drawable_get_depth = (Ptr_gdk_drawable_get_depth)QLibrary::resolve(GDK_PATH, 0, "gdk_drawable_get_depth");
+    QGtk::gdk_color_free = (Ptr_gdk_color_free)QLibrary::resolve(GDK_PATH, 0, "gdk_color_free");
+    QGtk::gdk_x11_window_set_user_time = (Ptr_gdk_x11_window_set_user_time)QLibrary::resolve(GDK_PATH, 0, "gdk_x11_window_set_user_time");
+    QGtk::gdk_x11_drawable_get_xid = (Ptr_gdk_x11_drawable_get_xid)QLibrary::resolve(GDK_PATH, 0, "gdk_x11_drawable_get_xid");
+    QGtk::gdk_x11_drawable_get_xdisplay = (Ptr_gdk_x11_drawable_get_xdisplay)QLibrary::resolve(GDK_PATH, 0, "gdk_x11_drawable_get_xdisplay");
 
     QGtk::gtk_widget_set_default_direction = (Ptr_gtk_widget_set_default_direction)QLibrary::resolve(GTK_PATH, 0, "gtk_widget_set_default_direction");
     QGtk::gtk_widget_modify_fg = (Ptr_gtk_widget_modify_color)QLibrary::resolve(GTK_PATH, 0, "gtk_widget_modify_fg");
@@ -315,10 +317,10 @@ static void resolveGtk()
     QGtk::gtk_widget_get_type =(Ptr_gtk_widget_get_type)QLibrary::resolve(GTK_PATH, 0, "gtk_widget_get_type");
     QGtk::gtk_rc_get_style_by_paths =(Ptr_gtk_rc_get_style_by_paths)QLibrary::resolve(GTK_PATH, 0, "gtk_rc_get_style_by_paths");
     QGtk::gtk_check_version =(Ptr_gtk_check_version)QLibrary::resolve(GTK_PATH, 0, "gtk_check_version");
-    QGtk::pango_font_description_get_size = (Ptr_pango_font_description_get_size)QLibrary::resolve(GTK_PATH, 0, "pango_font_description_get_size");
-    QGtk::pango_font_description_get_weight = (Ptr_pango_font_description_get_weight)QLibrary::resolve(GTK_PATH, 0, "pango_font_description_get_weight");
-    QGtk::pango_font_description_get_family = (Ptr_pango_font_description_get_family)QLibrary::resolve(GTK_PATH, 0, "pango_font_description_get_family");
-    QGtk::pango_font_description_get_style = (Ptr_pango_font_description_get_style)QLibrary::resolve(GTK_PATH, 0, "pango_font_description_get_style");
+    QGtk::pango_font_description_get_size = (Ptr_pango_font_description_get_size)QLibrary::resolve(QLS("pango-1.0"), 0, "pango_font_description_get_size");
+    QGtk::pango_font_description_get_weight = (Ptr_pango_font_description_get_weight)QLibrary::resolve(QLS("pango-1.0"), 0, "pango_font_description_get_weight");
+    QGtk::pango_font_description_get_family = (Ptr_pango_font_description_get_family)QLibrary::resolve(QLS("pango-1.0"), 0, "pango_font_description_get_family");
+    QGtk::pango_font_description_get_style = (Ptr_pango_font_description_get_style)QLibrary::resolve(QLS("pango-1.0"), 0, "pango_font_description_get_style");
 }
 
 void QGtk::cleanup_gtk_widgets()
