--- origsrc/qt-everywhere-opensource-src-4.6.2/src/multimedia/audio/audio.pri	2010-02-11 09:55:23.000000000 -0600
+++ src/qt-everywhere-opensource-src-4.6.2/src/multimedia/audio/audio.pri	2010-04-11 22:09:37.101561400 -0500
@@ -32,7 +32,7 @@ mac {
 
     LIBS += -framework ApplicationServices -framework CoreAudio -framework AudioUnit -framework AudioToolbox
 
-} else:win32 {
+} else:win32|cygwin-g++ {
 
     HEADERS += $$PWD/qaudioinput_win32_p.h $$PWD/qaudiooutput_win32_p.h $$PWD/qaudiodeviceinfo_win32_p.h
     SOURCES += $$PWD/qaudiodeviceinfo_win32_p.cpp \
