--- origsrc/qt-x11-opensource-src-4.4.0/src/corelib/plugin/qlibrary.cpp	2008-04-28 08:11:18.000000000 -0500
+++ src/qt-x11-opensource-src-4.4.0/src/corelib/plugin/qlibrary.cpp	2008-07-13 14:14:07.015625000 -0500
@@ -510,7 +510,7 @@
  */
 bool QLibrary::isLibrary(const QString &fileName)
 {
-#if defined(Q_OS_WIN32) || defined(Q_OS_WINCE)
+#if defined(Q_OS_WIN32) || defined(Q_OS_WINCE) || defined(Q_OS_CYGWIN)
     return fileName.endsWith(QLatin1String(".dll"));
 #else
     QString completeSuffix = QFileInfo(fileName).completeSuffix();
--- origsrc/qt-x11-opensource-src-4.6.2/src/corelib/plugin/qlibrary_unix.cpp
+++ src/qt-x11-opensource-src-4.6.2/src/corelib/plugin/qlibrary_unix.cpp
@@ -152,6 +152,13 @@
 
 #elif defined(Q_OS_SYMBIAN)
         suffixes << QLatin1String(".dll");
+#elif defined(Q_OS_CYGWIN)
+        prefixes << QLatin1String("cyg");
+        if (!fullVersion.isEmpty()) {
+            suffixes << QString::fromLatin1("-%1.dll").arg(fullVersion);
+        } else {
+            suffixes << QLatin1String(".dll");
+        }
 #else
         if (!fullVersion.isEmpty()) {
             suffixes << QString::fromLatin1(".so.%1").arg(fullVersion);
--- origsrc/qt-x11-opensource-src-4.5.1/src/corelib/io/qprocess_unix.cpp	2009-04-21 18:57:41.000000000 -0500
+++ src/qt-x11-opensource-src-4.5.1/src/corelib/io/qprocess_unix.cpp	2009-06-02 03:45:49.703763800 -0500
@@ -538,6 +538,8 @@ static char **_q_dupEnvironment(const QS
     // copy it over.
 #if defined(Q_OS_MAC)
     static const char libraryPath[] = "DYLD_LIBRARY_PATH";
+#elif defined(Q_OS_CYGWIN)
+    static const char libraryPath[] = "PATH";
 #else
     static const char libraryPath[] = "LD_LIBRARY_PATH";
 #endif
