--- origsrc/qt-everywhere-opensource-src-4.6.2/src/gui/styles/qgtkstyle_p.cpp	2010-02-11 09:55:23.000000000 -0600
+++ src/qt-everywhere-opensource-src-4.6.2/src/gui/styles/qgtkstyle_p.cpp	2010-05-03 23:45:29.042569700 -0500
@@ -306,6 +306,9 @@ void QGtkStylePrivate::resolveGtk() cons
 {
     // enforce the "0" suffix, so we'll open libgtk-x11-2.0.so.0
     QLibrary libgtk(QLS("gtk-x11-2.0"), 0, 0);
+    QLibrary libgdk(QLS("gdk-x11-2.0"), 0, 0);
+    QLibrary libgdkpixbuf(QLS("gdk_pixbuf-2.0"), 0, 0);
+    QLibrary libpango(QLS("pango-1.0"), 0, 0);
     libgtk.setLoadHints(QLibrary::ImprovedSearchHeuristics);
 
     gtk_init = (Ptr_gtk_init)libgtk.resolve("gtk_init");
@@ -328,20 +331,20 @@ void QGtkStylePrivate::resolveGtk() cons
     gtk_dialog_run = (Ptr_gtk_dialog_run)libgtk.resolve("gtk_dialog_run");
     gtk_file_chooser_set_filename = (Ptr_gtk_file_chooser_set_filename)libgtk.resolve("gtk_file_chooser_set_filename");
 
-    gdk_pixbuf_get_pixels = (Ptr_gdk_pixbuf_get_pixels)libgtk.resolve("gdk_pixbuf_get_pixels");
-    gdk_pixbuf_get_width = (Ptr_gdk_pixbuf_get_width)libgtk.resolve("gdk_pixbuf_get_width");
-    gdk_pixbuf_get_height = (Ptr_gdk_pixbuf_get_height)libgtk.resolve("gdk_pixbuf_get_height");
-    gdk_pixmap_new = (Ptr_gdk_pixmap_new)libgtk.resolve("gdk_pixmap_new");
-    gdk_pixbuf_new = (Ptr_gdk_pixbuf_new)libgtk.resolve("gdk_pixbuf_new");
-    gdk_pixbuf_get_from_drawable = (Ptr_gdk_pixbuf_get_from_drawable)libgtk.resolve("gdk_pixbuf_get_from_drawable");
-    gdk_draw_rectangle = (Ptr_gdk_draw_rectangle)libgtk.resolve("gdk_draw_rectangle");
-    gdk_pixbuf_unref = (Ptr_gdk_pixbuf_unref)libgtk.resolve("gdk_pixbuf_unref");
-    gdk_drawable_unref = (Ptr_gdk_drawable_unref)libgtk.resolve("gdk_drawable_unref");
-    gdk_drawable_get_depth = (Ptr_gdk_drawable_get_depth)libgtk.resolve("gdk_drawable_get_depth");
-    gdk_color_free = (Ptr_gdk_color_free)libgtk.resolve("gdk_color_free");
-    gdk_x11_window_set_user_time = (Ptr_gdk_x11_window_set_user_time)libgtk.resolve("gdk_x11_window_set_user_time");
-    gdk_x11_drawable_get_xid = (Ptr_gdk_x11_drawable_get_xid)libgtk.resolve("gdk_x11_drawable_get_xid");
-    gdk_x11_drawable_get_xdisplay = (Ptr_gdk_x11_drawable_get_xdisplay)libgtk.resolve("gdk_x11_drawable_get_xdisplay");
+    gdk_pixbuf_get_pixels = (Ptr_gdk_pixbuf_get_pixels)libgdkpixbuf.resolve("gdk_pixbuf_get_pixels");
+    gdk_pixbuf_get_width = (Ptr_gdk_pixbuf_get_width)libgdkpixbuf.resolve("gdk_pixbuf_get_width");
+    gdk_pixbuf_get_height = (Ptr_gdk_pixbuf_get_height)libgdkpixbuf.resolve("gdk_pixbuf_get_height");
+    gdk_pixmap_new = (Ptr_gdk_pixmap_new)libgdk.resolve("gdk_pixmap_new");
+    gdk_pixbuf_new = (Ptr_gdk_pixbuf_new)libgdkpixbuf.resolve("gdk_pixbuf_new");
+    gdk_pixbuf_get_from_drawable = (Ptr_gdk_pixbuf_get_from_drawable)libgdk.resolve("gdk_pixbuf_get_from_drawable");
+    gdk_draw_rectangle = (Ptr_gdk_draw_rectangle)libgdk.resolve("gdk_draw_rectangle");
+    gdk_pixbuf_unref = (Ptr_gdk_pixbuf_unref)libgdkpixbuf.resolve("gdk_pixbuf_unref");
+    gdk_drawable_unref = (Ptr_gdk_drawable_unref)libgdk.resolve("gdk_drawable_unref");
+    gdk_drawable_get_depth = (Ptr_gdk_drawable_get_depth)libgdk.resolve("gdk_drawable_get_depth");
+    gdk_color_free = (Ptr_gdk_color_free)libgdk.resolve("gdk_color_free");
+    gdk_x11_window_set_user_time = (Ptr_gdk_x11_window_set_user_time)libgdk.resolve("gdk_x11_window_set_user_time");
+    gdk_x11_drawable_get_xid = (Ptr_gdk_x11_drawable_get_xid)libgdk.resolve("gdk_x11_drawable_get_xid");
+    gdk_x11_drawable_get_xdisplay = (Ptr_gdk_x11_drawable_get_xdisplay)libgdk.resolve("gdk_x11_drawable_get_xdisplay");
 
     gtk_widget_set_default_direction = (Ptr_gtk_widget_set_default_direction)libgtk.resolve("gtk_widget_set_default_direction");
     gtk_widget_modify_fg = (Ptr_gtk_widget_modify_color)libgtk.resolve("gtk_widget_modify_fg");
@@ -420,10 +423,10 @@ void QGtkStylePrivate::resolveGtk() cons
     gtk_rc_get_style_by_paths =(Ptr_gtk_rc_get_style_by_paths)libgtk.resolve("gtk_rc_get_style_by_paths");
     gtk_check_version =(Ptr_gtk_check_version)libgtk.resolve("gtk_check_version");
     gtk_border_free =(Ptr_gtk_border_free)libgtk.resolve("gtk_border_free");
-    pango_font_description_get_size = (Ptr_pango_font_description_get_size)libgtk.resolve("pango_font_description_get_size");
-    pango_font_description_get_weight = (Ptr_pango_font_description_get_weight)libgtk.resolve("pango_font_description_get_weight");
-    pango_font_description_get_family = (Ptr_pango_font_description_get_family)libgtk.resolve("pango_font_description_get_family");
-    pango_font_description_get_style = (Ptr_pango_font_description_get_style)libgtk.resolve("pango_font_description_get_style");
+    pango_font_description_get_size = (Ptr_pango_font_description_get_size)libpango.resolve("pango_font_description_get_size");
+    pango_font_description_get_weight = (Ptr_pango_font_description_get_weight)libpango.resolve("pango_font_description_get_weight");
+    pango_font_description_get_family = (Ptr_pango_font_description_get_family)libpango.resolve("pango_font_description_get_family");
+    pango_font_description_get_style = (Ptr_pango_font_description_get_style)libpango.resolve("pango_font_description_get_style");
 
     gnome_icon_lookup_sync = (Ptr_gnome_icon_lookup_sync)QLibrary::resolve(QLS("gnomeui-2"), 0, "gnome_icon_lookup_sync");
     gnome_vfs_init= (Ptr_gnome_vfs_init)QLibrary::resolve(QLS("gnomevfs-2"), 0, "gnome_vfs_init");
--- origsrc/qt-everywhere-opensource-src-4.6.2/src/gui/styles/qgtkstyle_p.h	2010-02-11 09:55:22.000000000 -0600
+++ src/qt-everywhere-opensource-src-4.6.2/src/gui/styles/qgtkstyle_p.h	2010-05-03 22:46:27.167986100 -0500
@@ -63,6 +63,9 @@
 
 #undef signals // Collides with GTK stymbols
 #include <gtk/gtk.h>
+#ifdef G_PLATFORM_WIN32
+#undef gtk_init
+#endif
 
 typedef unsigned long XID;
 
