--- origsrc/qt-everywhere-opensource-src-4.7.1/src/corelib/global/qglobal.cpp	2010-11-05 20:55:18.000000000 -0500
+++ src/qt-everywhere-opensource-src-4.7.1/src/corelib/global/qglobal.cpp	2010-12-25 21:51:24.198696800 -0600
@@ -1701,7 +1701,7 @@ static QSysInfo::MacVersion macVersion()
 }
 const QSysInfo::MacVersion QSysInfo::MacintoshVersion = macVersion();
 
-#elif defined(Q_OS_WIN32) || defined(Q_OS_CYGWIN) || defined(Q_OS_WINCE)
+#elif defined(Q_OS_WIN32) || defined(Q_OS_WINCE)
 
 QT_BEGIN_INCLUDE_NAMESPACE
 #include "qt_windows.h"
--- origsrc/qt-everywhere-opensource-src-4.7.1/src/corelib/global/qnamespace.h	2010-11-05 20:55:18.000000000 -0500
+++ src/qt-everywhere-opensource-src-4.7.1/src/corelib/global/qnamespace.h	2010-12-25 21:51:24.183096800 -0600
@@ -1205,7 +1205,7 @@ public:
     };
 #endif // Q_OS_MAC
 
-#if defined(Q_OS_WIN) || defined(Q_OS_CYGWIN)
+#if defined(Q_OS_WIN)
 #ifndef qdoc
     typedef int WindowsVersion;
 
--- origsrc/qt-everywhere-opensource-src-4.8.1/src/corelib/io/qfilesystemiterator_p.h	2012-03-14 09:01:18.000000000 -0500
+++ src/qt-everywhere-opensource-src-4.8.1/src/corelib/io/qfilesystemiterator_p.h	2012-04-24 00:48:07.269358700 -0500
@@ -103,7 +103,7 @@ private:
 #else
     QT_DIR *dir;
     QT_DIRENT *dirEntry;
-#if defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(Q_OS_CYGWIN)
+#if defined(_POSIX_THREAD_SAFE_FUNCTIONS)
     // for readdir_r
     QScopedPointer<QT_DIRENT, QScopedPointerPodDeleter> mt_file;
 #endif
--- origsrc/qt-everywhere-opensource-src-4.8.1/src/corelib/io/qfilesystemiterator_unix.cpp	2012-03-14 09:01:18.000000000 -0500
+++ src/qt-everywhere-opensource-src-4.8.1/src/corelib/io/qfilesystemiterator_unix.cpp	2012-04-24 00:36:19.153856800 -0500
@@ -70,1 +70,1 @@ QFileSystemIterator::QFileSystemIterator
-#if defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(Q_OS_CYGWIN)
+#if defined(_POSIX_THREAD_SAFE_FUNCTIONS)
@@ -96,1 +96,1 @@ bool QFileSystemIterator::advance(QFileS
-#elif defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(Q_OS_CYGWIN)
+#elif defined(_POSIX_THREAD_SAFE_FUNCTIONS)
--- origsrc/qt-everywhere-opensource-src-4.7.1/src/gui/kernel/qapplication.h	2010-11-05 20:55:16.000000000 -0500
+++ src/qt-everywhere-opensource-src-4.7.1/src/gui/kernel/qapplication.h	2010-12-25 21:51:24.198696800 -0600
@@ -330,7 +330,7 @@ public:
     enum { NormalColors = NormalColor, CustomColors = CustomColor };
     static inline QT3_SUPPORT ColorMode colorMode() { return static_cast<ColorMode>(colorSpec()); }
     static inline QT3_SUPPORT void setColorMode(ColorMode mode) { setColorSpec(int(mode)); }
-#if defined(Q_OS_WIN32) || defined(Q_OS_CYGWIN)
+#if defined(Q_OS_WIN32)
     static QT3_SUPPORT Qt::WindowsVersion winVersion() { return (Qt::WindowsVersion)QSysInfo::WindowsVersion; }
 #endif
 #if defined(Q_OS_MAC)
--- origsrc/qt-everywhere-opensource-src-4.7.1/src/gui/kernel/qapplication_p.h	2010-11-05 20:55:17.000000000 -0500
+++ src/qt-everywhere-opensource-src-4.7.1/src/gui/kernel/qapplication_p.h	2010-12-25 21:51:24.214296800 -0600
@@ -93,7 +93,7 @@ extern bool qt_is_gui_used;
 extern QClipboard *qt_clipboard;
 #endif
 
-#if defined (Q_OS_WIN32) || defined (Q_OS_CYGWIN) || defined(Q_OS_WINCE)
+#if defined (Q_OS_WIN32) || defined(Q_OS_WINCE)
 extern QSysInfo::WinVersion qt_winver;
 enum { QT_TABLET_NPACKETQSIZE = 128 };
 # ifdef Q_OS_WINCE
--- origsrc/qt-everywhere-opensource-src-4.7.1/src/qt3support/network/q3dns.cpp	2010-11-05 20:55:17.000000000 -0500
+++ src/qt-everywhere-opensource-src-4.7.1/src/qt3support/network/q3dns.cpp	2010-12-25 21:51:11.141474000 -0600
@@ -42,7 +42,7 @@
 #include "qplatformdefs.h"
 #include "qbytearray.h"
 #include <private/qsystemlibrary_p.h>
-#if defined(Q_OS_WIN32) || defined(Q_OS_WINCE) || defined(Q_OS_CYGWIN)
+#if defined(Q_OS_WIN32) || defined(Q_OS_WINCE)
 # include "qt_windows.h"
 #else
 # include <sys/types.h>
@@ -2205,7 +2205,7 @@ void Q3Dns::connectNotify( const char *s
 }
 #endif
 
-#if defined(Q_OS_WIN32) || defined(Q_OS_CYGWIN)
+#if defined(Q_OS_WIN32)
 
 #if defined(Q_DNS_SYNCHRONOUS)
 void Q3Dns::doSynchronousLookup()
